### PR TITLE
[MSHARED-1071] Drop maven 3.0 compatibility, Maven 3.2.5, Injects

### PR DESCRIPTION
**Pull request by gnodet**

### CODE REVIEW

**Review by slachiewicz — CHANGES_REQUESTED** [review]
That was rejected some time ago

**Comment by gnodet** [comment]
> That was rejected some time ago

What was rejected exactly ? The drop of legacy maven 3.0.x support ?

**Comment by slachiewicz** [comment]
Yes #6

**Comment by gnodet** [comment]
> Yes #6

Does this mean this artifact should be deprecated and somehow merged into maven-core ? Same as maven-artifact-transfer / maven-common-artifact-filters ?

**Comment by gnodet** [comment]
> > Yes #6
> 
> Does this mean this artifact should be deprecated and somehow merged into maven-core ? Same as maven-artifact-transfer / maven-common-artifact-filters ?

Something like https://github.com/gnodet/maven/tree/mat ?

**Review by cstamas — APPROVED** [review]
LGTM

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]
Jira issue will be appreciated.

**Review by michael-o — CHANGES_REQUESTED** [review]
Look here: https://cwiki.apache.org/confluence/display/MAVEN/Maven+Ecosystem+Cleanup#MavenEcosystemCleanup-ResolverandMaven
Maven 3.1.1 comes with Aether not Resolver, does this matter?

**Comment by gnodet** [comment]
> Look here: https://cwiki.apache.org/confluence/display/MAVEN/Maven+Ecosystem+Cleanup#MavenEcosystemCleanup-ResolverandMaven
> Maven 3.1.1 comes with Aether not Resolver, does this matter?

@michael-o I would prefer aligning everything on using resolver, but that would mean moving the requirements to Maven 3.5.x.

**Comment by michael-o** [comment]
> > Look here: https://cwiki.apache.org/confluence/display/MAVEN/Maven+Ecosystem+Cleanup#MavenEcosystemCleanup-ResolverandMaven
> > Maven 3.1.1 comes with Aether not Resolver, does this matter?
> 
> @michael-o I would prefer aligning everything on using resolver, but that would mean moving the requirements to Maven 3.5.x.

I don't disagree, in fact I would prefer to move ALL components to 3.5.4 after Maven 4.0.0-alpha-1 has been released, but consider that this is used in many other components, thus forcing them to update as well. :-(

**Comment by gnodet** [comment]
> > > Look here: https://cwiki.apache.org/confluence/display/MAVEN/Maven+Ecosystem+Cleanup#MavenEcosystemCleanup-ResolverandMaven
> > > Maven 3.1.1 comes with Aether not Resolver, does this matter?
> > 
> > 
> > @michael-o I would prefer aligning everything on using resolver, but that would mean moving the requirements to Maven 3.5.x.
> 
> I don't disagree, in fact I would prefer to move ALL components to 3.5.4 after Maven 4.0.0-alpha-1 has been released, but consider that this is used in many other components, thus forcing them to update as well. :-(

If we start defining a new API to hide the resolver, this component may simply be deprecated at some point.

**Comment by michael-o** [comment]
> > > > Look here: https://cwiki.apache.org/confluence/display/MAVEN/Maven+Ecosystem+Cleanup#MavenEcosystemCleanup-ResolverandMaven
> > > > Maven 3.1.1 comes with Aether not Resolver, does this matter?
> > > 
> > > 
> > > @michael-o I would prefer aligning everything on using resolver, but that would mean moving the requirements to Maven 3.5.x.
> > 
> > 
> > I don't disagree, in fact I would prefer to move ALL components to 3.5.4 after Maven 4.0.0-alpha-1 has been released, but consider that this is used in many other components, thus forcing them to update as well. :-(
> 
> If we start defining a new API to hide the resolver, this component may simply be deprecated at some point.

Who's actually using this component to estimate the impact?

**Comment by slawekjaranowski** [comment]
> Who's actually using this component to estimate the impact?

more or less usage report: https://mvnrepository.com/artifact/org.apache.maven.shared/maven-dependency-tree/usages

**Comment by slawekjaranowski** [comment]
And issue https://issues.apache.org/jira/browse/MSHARED-1071

**Review by slawekjaranowski — APPROVED** [review]
Connect with issue and go ...

**Comment by slawekjaranowski** [comment]
I think that it is still valuable, @gnodet  could you rebase and resolve conflicts ... if you don't have time I can take care about it.

File: pom.xml [hunk 1]
```diff
@@ -55,7 +55,9 @@
   </distributionManagement>
 
   <properties>
-    <javaVersion>8</javaVersion>
+	  <javaVersion>8</javaVersion>
```
Comment (michael-o) [inline pom.xml:58]: Indent broken

File: pom.xml [hunk 2]
```diff
@@ -69,51 +71,38 @@
...
-          <groupId>org.eclipse.aether</groupId>
-          <artifactId>aether-api</artifactId>
-        </exclusion>
-      </exclusions>
+      <version>${resolverVersion}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <version>1.7.33</version>
```
Comment (michael-o) [inline pom.xml:91]: 36?

**Review by michael-o — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (slawekjaranowski) [reply pom.xml]: tab 😄

**Review by slachiewicz — APPROVED** [review]

**Comment by slawekjaranowski** [comment]
Any remarks for it?
I'm going to merge it in a few days.

File: pom.xml [hunk 3]
```diff
@@ -56,6 +56,8 @@
 
   <properties>
     <javaVersion>8</javaVersion>
+    <maven.version>3.2.5</maven.version>
```
Comment (michael-o) [inline pom.xml:59]: This should really be `mavenVersion`.

**Review by michael-o — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 3 above)
Comment (slawekjaranowski) [reply pom.xml]: done

**Review by slachiewicz — APPROVED** [review]

**Comment by slachiewicz** [comment]
docs also should be updated:

> Component entry point is [DependencyGraphBuilder](https://maven.apache.org/shared/maven-dependency-tree/apidocs/org/apache/maven/shared/dependency/graph/DependencyGraphBuilder.html), which will delegate to appropriate implementation for either Maven 2, Maven 3.0.x or Maven 3.1.x.

**Review by slachiewicz — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #86

**Pull request merged**